security/src/main/java/io/micronaut/security/config/InterceptUrlMapConverter.java

```diff
@@ -40,12 +43,31 @@ public class InterceptUrlMapConverter implements TypeConverter<Map<String, Objec
     private static final String HTTP_METHOD = "http-method";
 
     private final ConversionService conversionService;
+    private final ServerContextPathProvider serverContextPathProvider;
+
+    private final boolean interceptUrlMapPrependPatternWithContextPath;
 
     /**
-     * @param conversionService The conversion service
+     * @param conversionService     The conversion service
+     * @deprecated Use {@link InterceptUrlMapConverter(ConversionService,ServerContextPathProvider)} instead.
      */
+    @Deprecated
     InterceptUrlMapConverter(ConversionService conversionService) {
+        this(conversionService, () -> null, false);
+    }
+
+    /**
+     * @param conversionService     The conversion service
+     * @param serverContextPathProvider context path provider
+     * @param interceptUrlMapPrependPatternWithContextPath Whether the intercept URL patterns should be prepended with context path if defined.
+     */
+    @Inject
+    InterceptUrlMapConverter(ConversionService conversionService,
+                             ServerContextPathProvider serverContextPathProvider,
+                             @Value("${" + SecurityConfigurationProperties.PREFIX + ".intercept-url-map-prepend-pattern-with-context-path:" + SecurityConfigurationProperties.DEFAULT_INTERCEPT_URL_MAP_PREPEND_PATTERN_WITH_CONTEXT_PATH + "}") boolean interceptUrlMapPrependPatternWithContextPath) {
```

Comment:
Why does this not just inject the SecurityConfiguration?

Reply:
There is an instantiation error. I think it is because there is a circular dependency. `SecurityConfigurationProperties` the implementation of `SecurityConfiguration` uses the `InterceptUrlMapConverter`